base-lite/pom.xml

```diff
@@ -0,0 +1,97 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--~
+  ~ Copyright 2017 Confluent Inc.
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~ http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  ~-->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>io.confluent</groupId>
+        <artifactId>common-docker</artifactId>
+        <version>7.4.0-0</version>
+    </parent>
+
+    <packaging>pom</packaging>
+
+    <artifactId>cp-base-lite</artifactId>
+
+    <description>Base for new Confluent lightweight Docker images</description>
+    <name>${project.artifactId}</name>
+
+    <properties>
+        <docker.skip-build>false</docker.skip-build>
+        <docker.skip-test>false</docker.skip-test>
+        <docker.pull-image>true</docker.pull-image>
+    </properties>
+
+    <dependencies>
```

Comment:
@elismaga This image would be containerised form of AK with no extra add ons. Any other runtime dependencies would be done in installation tools Ansible/CFK. We will work on the later part to get that done efficiently

Reply:
@nsharma-git This isn't an image for Kafka. Also, we don't build an image with AK. This is a base image that needs to contain libraries that are common dependencies for all services. It doesn't make sense to add all these dependencies during installation with Ansible as it will slow down the startup time and be a negative impact on the user experience. It will also make it harder to maintain, as it is easier to update the versions if these libraries when they are defined in a central place like the base image.

I am saying these are required and not add on features in my opinion. Have you checked to see what will not work when you remove these libraries? It seems like the logging won't work if you remove logredactor and slf4j. I wouldn't consider logging to be an add-on feature, but a required feature.

Reply:
We are aware of the possible runtime delay and user experience. AFAIK, Ansible does install the softwares at runtime. Moreover Ansible is not based on the docker images anyway. CFK uses our ce-kafka image, which is not going to get impacted by this change. If at all we get this base for all images, in that case If something is needed, we can get them while building the ce-kafka images. That way we can target our both group of users. We can also think of adding another base layer. 

This is certainly going to increase the maintenance, however, this is important from product and placement of Kafka against the new competitive solutions in market.
With this change, we are trying to boost the dev first experience with super light and fast version of Kafka,. The image has a completely different set of target users.

Log redactor feature we added sometime back and its not part of AK. Same way, slf4j is packaged with AK. It would be available in the the Kafka lite image. We along with other field team has tried Kafka lite image based on this base image and it works fine. We will be adding more tests in kafka lite image for the integration points anyway.

Please let me know if this clarifies your concerns. Thanks!